--- hybrid-v35_64-nodebug-pcoem-6_30_223_248.orig/src/wl/sys/wl_cfg80211_hybrid.c	2015-04-23 10:54:13.313236188 -0400
+++ hybrid-v35_64-nodebug-pcoem-6_30_223_248/src/wl/sys/wl_cfg80211_hybrid.c	2015-04-23 10:54:44.769217013 -0400
@@ -1819,7 +1819,7 @@
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 11, 0)
 static const struct wiphy_wowlan_support wl_wowlan_support = {
 #else
-static struct wiphy_wowlan_support wl_wowlan_support = {
+static const struct wiphy_wowlan_support wl_wowlan_support = {
 #endif 
 	.flags = WIPHY_WOWLAN_MAGIC_PKT
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 1, 0)
@@ -1885,7 +1885,7 @@
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 11, 0)
 	wdev->wiphy->wowlan = &wl_wowlan_support;
 #else
-	wdev->wiphy->wowlan = wl_wowlan_support;
+	wdev->wiphy->wowlan = &wl_wowlan_support;
 #endif 
 #endif 
 #endif 
@@ -2059,7 +2059,7 @@
 	u32 event = EVENT_TYPE(e);
 	u16 flags = EVENT_FLAGS(e);
 	u32 status = EVENT_STATUS(e);
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 15, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 10, 0)
 	struct wl_bss_info *bi;
 	u16 bss_info_channel;
 	struct ieee80211_channel *channel;
@@ -2100,7 +2100,7 @@
 			wl_get_assoc_ies(wl);
 			memcpy(&wl->bssid, &e->addr, ETHER_ADDR_LEN);
 			wl_update_bss_info(wl);
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 15, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 10, 0)
 			bi = (struct wl_bss_info *)(wl->extra_buf + 4);
 			bss_info_channel = bi->ctl_ch ? bi->ctl_ch : CHSPEC_CHANNEL(bi->chanspec);
 
